Cesium3DTilesSelection/src/CompositeContent.cpp

```diff
@@ -30,6 +34,64 @@ static_assert(sizeof(InnerHeader) == 12);
 namespace Cesium3DTilesSelection {
 
 namespace {
+
+class DerivedInnerResponse : public CesiumAsync::IAssetResponse {
+private:
+  const IAssetResponse* pOriginalResponse;
+  gsl::span<const std::byte> derivedData;
+
+public:
+  DerivedInnerResponse(
+      const IAssetResponse* pOriginalResponse_,
+      const gsl::span<const std::byte> derivedData_)
+      : pOriginalResponse(pOriginalResponse_), derivedData(derivedData_) {}
+
+  uint16_t statusCode() const override {
+    return pOriginalResponse->statusCode();
+  }
+
+  std::string contentType() const override {
+    return pOriginalResponse->contentType();
```

Comment:
In the original implementation, the `contentType` for the nested content was set to empty string. I'm not sure if that was actually important or not, though.